Sources/NetworkProtection/Status/NetworkProtectionStatusReporter.swift

```diff
@@ -57,77 +57,40 @@ public struct NetworkProtectionStatusServerInfo: Codable, Equatable {
 ///
 public final class DefaultNetworkProtectionStatusReporter: NetworkProtectionStatusReporter {
 
-    // MARK: - Logging
-
-    /// The logger that this object will use for errors that are handled by this class.
-    ///
-    private let log: OSLog
-
     // MARK: - Notifications
 
     private let distributedNotificationCenter: DistributedNotificationCenter
 
-    // MARK: - Notifications: Observation Tokens
-
-    private var cancellables = Set<AnyCancellable>()
-
     // MARK: - Publishers
 
-    private let statusObserver: ConnectionStatusObserver
-    public var statusPublisher: CurrentValueSubject<ConnectionStatus, Never> {
-        statusObserver.publisher
-    }
-    public let connectivityIssuesPublisher = CurrentValueSubject<Bool, Never>(false)
-    private let serverInfoObserver: ConnectionServerInfoObserver
-    public var serverInfoPublisher: CurrentValueSubject<NetworkProtectionStatusServerInfo, Never> {
-        serverInfoObserver.publisher
-    }
-    private let connectionErrorObserver: ConnectionErrorObserver
-    public var connectionErrorPublisher: CurrentValueSubject<String?, Never> {
-        connectionErrorObserver.publisher
-    }
-    public let controllerErrorMessagePublisher = CurrentValueSubject<String?, Never>(nil)
+    public let statusObserver: ConnectionStatusObserver
+    public let serverInfoObserver: ConnectionServerInfoObserver
+    public let connectionErrorObserver: ConnectionErrorObserver
+    public let connectivityIssuesObserver: ConnectivityIssueObserver
+    public let controllerErrorMessageObserver: ControllerErrorMesssageObserver
 
     // MARK: - Init & deinit
 
     public init(statusObserver: ConnectionStatusObserver,
                 serverInfoObserver: ConnectionServerInfoObserver,
                 connectionErrorObserver: ConnectionErrorObserver,
-                distributedNotificationCenter: DistributedNotificationCenter = .default(),
-                log: OSLog = .networkProtectionStatusReporterLog) {
+                connectivityIssuesObserver: ConnectivityIssueObserver,
+                controllerErrorMessageObserver: ControllerErrorMesssageObserver,
+                distributedNotificationCenter: DistributedNotificationCenter = .default()) {
 
         self.statusObserver = statusObserver
         self.serverInfoObserver = serverInfoObserver
         self.connectionErrorObserver = connectionErrorObserver
+        self.connectivityIssuesObserver = connectivityIssuesObserver
+        self.controllerErrorMessageObserver = controllerErrorMessageObserver
         self.distributedNotificationCenter = distributedNotificationCenter
-        self.log = log
 
         start()
     }
 
     // MARK: - Starting & Stopping
 
     private func start() {
-        distributedNotificationCenter.publisher(for: .controllerErrorChanged).sink { [weak self] notification in
```

Comment:
This and the following statement were moved to `ConnectivityIssueObserver` https://github.com/duckduckgo/BrowserServicesKit/pull/445/files#r1285476586